glean-core/ffi/src/lib.rs

```diff
@@ -198,16 +198,70 @@ pub extern "C" fn glean_enable_logging() {
     }
 }
 
-#[no_mangle]
-pub extern "C" fn glean_initialize(
-    data_dir: FfiStr,
-    application_id: FfiStr,
+#[repr(C)]
+pub struct FfiConfiguration<'a> {
+    data_dir: FfiStr<'a>,
+    package_name: FfiStr<'a>,
     upload_enabled: u8,
-) -> u64 {
+    max_events: Option<&'a i64>,
+}
+
+#[derive(Debug)]
+pub struct GleanConfiguration {
```

Comment:
I like the way it works here. I'd recommend moving this out of the FFI layer and use this object for initializing glean, i.e. `Glean::new(config)` rather than passing in all the things. That will save us a bit of time in case somebody wants to start to consume the Rust bits.

Reply:
Yup, that would be the plan